### DISCUSSION TITLE
cannot export large snapshot

### DISCUSSION THREAD

**Discussion by wangzhankun** [coddingtonbear/obsidian-web#173] [Exotic Browser Support]
### Describe the bug

Firstly, I have met the same issue with #120 , but after I have upgrade the `obsidian-local-rest-api` to 2.2.1, the issue has been resolved.

But when I try to capture [the web page](https://www.kernel.org/doc/Documentation/cgroup-v2.txt), obsidian-web extension says all is done. But there is no such page have been captured in my obsidian vault. I have test another website, all function is ok.

### To Reproduce

1. go to https://www.kernel.org/doc/Documentation/cgroup-v2.txt
2.  capture page snapshot
<img width="459" alt="image" src="https://github.com/coddingtonbear/obsidian-web/assets/35724924/d7697ddb-8986-43ab-ad9f-cd379b125a33">

3. All done! But no content found.
<img width="466" alt="image" src="https://github.com/coddingtonbear/obsidian-web/assets/35724924/e5789b44-4657-4be8-accf-c49157bf98b0">

### Operating System

ubuntu 22.04

### Browser

microsoft edge

### Browser Version

122.0.2365.52

**Comment by coddingtonbear** [comment]
I believe this might be a duplicate of #172 .  Unfortunately, I haven't been able to reproduce this issue locally, but if you join that discussion on #172 maybe you can find a way of helping that discussion along.

**Reply by wangzhankun** [reply]
No, it's not same as #172. All work fine when capture a page which is small. I only encountered this problem when clipping this page [https://www.kernel.org/doc/Documentation/cgroup-v2.txt](https://www.kernel.org/doc/Documentation/cgroup-v2.txt).  In addition, no matter how many times I repeat the clip operation, I cannot successfully obtain this page.
I have pressed `F12` to try to find any log that the extension output, but nothing found in the console.

**Reply by coddingtonbear** [reply]
The problem in that particular case is literally that there is no page title set on that page, and the template you're using assumes that there is.  Fortunately, such situations are quite rare on the internet these days, but unfortunately the only option I have to offer you around this is either:

1. You modify the "API URL" field from "View Request Details" before submitting such files.  I'm considering adding some validation before submission that will check to verify that your `API URL` has a valid filename (that is: isn't just an extension like `.md`), but that will require some thought to make sure it's a good idea.  In such a case, you'd just get an error message instead of the "All Done" message, though, so you'd still have to fix it yourself.  Importantly: the "All Done" message _is_ correct -- the file was written, it's just named `.md` and might be hidden from display since files starting with `.` are usually hidden.
2. You alter your template such that your "API URL" field is populated with something other than just `{{ filename.page_title }}` since `page.title` isn't always available.  I'm not sure what might be a good choice, but perhaps the current date and time might be sufficient?

**Comment by wangzhankun** [comment]
I think the reason is the extension cannot get the filename:
![image](https://github.com/coddingtonbear/obsidian-web/assets/35724924/8d613fff-4157-43cc-8bb7-2b85bd0be268)

**Reply by coddingtonbear** [reply]
If so, that means one of two things:

- That this *is* the same issue as #172 -- the root cause of #172 is that we're not calculating a filename for some still unknown reason.
- That there isn't a `page.title` for us to get a filename from.  This would be a problem you'd have to resolve, unfortunately, in your template or by modifying the request details.

I _suspect_ that it's the former (that this is the same issue as #172), but if you think it's instead a problem with your template, the resolution is that you just need to manually enter a different API URL in such cases.  If your `page.title` either contains no data (because the page has no title) or `page.title` contains only characters that are not filename-safe, there isn't much help I can offer you except suggesting that if you run into this often perhaps you could include other data in your filename so that there would always be content to render there.

**Reply by wangzhankun** [reply]
Is it possible to add a logic to fill in a default value when `page.title` cannot be obtained? For example, the content after the last `/` in the URL is used as the default file name.

**Reply by coddingtonbear** [reply]
I'm afraid even that isn't necessarily sufficient!  What if there's nothing after the `/`?  What if there isn't even a path before that? What if it's not even the `page.title` that we're using for generating the filename, but some other context we gather from the page?  Should we have defaults in all of those cases, too?  

Setting aside the fact that it's not all that clear what a good "default" would be, it's also very much the sort of thing that people might have opinions about so I'm extremely hesitant about the idea of having any kind of "default" that's set across-the-board and I'd rather that each user user configure whatever behavior they desire by configuring their templates as necessary.

**Reply by coddingtonbear** [reply]
& to be clear, you could adjust your template to handle this by doing something as simple as replacing `{{ filename page.title }}` with something like `{{#if page.title}}{{ filename page.title }}{{ else }}{{ uuid }}{{/if}}.md`.  In such a case, if the `page.title` didn't have a value, we'd instead just generate a random UUID.

**Reply by wangzhankun** [reply]
Thanks
